lib/index.js

```diff
@@ -974,3 +984,23 @@ exports.shallow = function (source) {
 
     return target;
 };
+
+
+exports.promiseWrap = function (bind, method, args, transform) {
+
+    return new Promise((resolve, reject) => {
+
+        const callback = function (error, result) {
+
+            if (Array.isArray(transform)) {
+                result = exports.transform(Array.from(arguments), transform);
+                error instanceof Error ? reject(result) : resolve(result);
```

Comment:
Not sure it's a use case that should be covered. Receiving an object in a catch is quite unexpected, this would be a weird resulting promise.